src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -48,6 +51,9 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
 
         CommandResult commandResult;
         Command command = addressBookParser.parseCommand(commandText);
+        if (command.getCommandWord() != "undo") {
+            pastCommands.add(command);
```

Comment:
It would be better to save the past commands only after it was successfully executed. Otherwise, the app may not show an error message if the user issues an `undo` command after a failed command (e.g., `delete 99`).

Reply:
thnks will update